### ISSUE TITLE
job submission script for parallel job

### ISSUE THREAD

**Issue by RayAmini** [ru-ccmt/eDMFT#3]
Hi!

I started a new issue for this:

> Hi,
> Is there any proper way to run the parallel job of DMFT.
> The serial job is going fine but when I try to run in parallel the code is exiting some kind of python error.
> Is there any proper job submission script for parallel job?
> 
> Thank you.

There are examples in the [MnO tutorial](http://hauleweb.rutgers.edu/tutorials/Tutorial1a.html)

You may need to adapt these to your own cluster environment. If you need further help, post also your job script and some output.

**Comment by SreenivasIKST** [comment]
Hi RayAmini,

Thank you for creating a new issue.  
I tried the parallel job with 4 processors.
I have created .machines file as below
1:localhost
1:localhost
1:localhost
1:localhost
granularity:1
extrafine:1

I used the job script as shown below.

#!/bin/bash
#PBS -l nodes=master:ppn=4
#PBS -e test.err
#PBS -o test.out
#PBS -N test_wien

source /etc/profile.d/modules.sh

cd $PBS_O_WORKDIR

module purge
module load intel-2022.1/compiler
module load intel-2022.1/intelmpi
module load edmft-reddy

cd $PBS_O_WORKDIR
echo "mpirun -np 4 -x OMP_NUM_THREADS=1" > mpi_prefix.dat
echo "mpirun -np 4 -x OMP_NUM_THREADS=1" > mpi_prefix.dat2

$WIEN_DMFT_ROOT/run_dmft.py >& nohup.dat

eval $run 1> >(tee ${PBS_JOBID}.${PBS_JOBNAME}.out) 2> >(tee -a ${PBS_JOBID}.${PBS_JOBNAME}.out)

Here are the error messages.

LAPW0 END
 LAPW1 END
[1]    Done                          ( cd $PWD; $t $exe ${def}_$loop.def; rm -f .lock_$lockfile[$p] ) >> .time1_$loop
 LAPW1 END
 LAPW1 END
 LAPW1 END
[4]  - Done                          ( cd $PWD; $t $exe ${def}_$loop.def; rm -f .lock_$lockfile[$p] ) >> .time1_$loop
[3]  + Done                          ( cd $PWD; $t $exe ${def}_$loop.def; rm -f .lock_$lockfile[$p] ) >> .time1_$loop
[2]  + Done                          ( cd $PWD; $t $exe ${def}_$loop.def; rm -f .lock_$lockfile[$p] ) >> .time1_$loop
DmftEnvironment: mpi_prefix.dat exists -- running in parallel mode.
   mpirun -np 4 -x OMP_NUM_THREADS=1
DmftEnvironment: mpi_prefix.dat2 is different from mpi_prefix.dat -- running dmft1/dmft2 in parallel mode.
   mpirun -np 4 -x OMP_NUM_THREADS=1
Traceback (most recent call last):
  File "/home/reddy/soft/CODES/eDMFT-intel/bin/run_dmft.py", line 1007, in <module>
    nc = ConvertFromLatticeToImp(nd, imp2latt)
  File "/home/reddy/soft/CODES/eDMFT-intel/bin/imp2lattc.py", line 69, in ConvertFromLatticeToImp
    ndc=[ nd[a-1] for a in imp2latt[ic] ]
  File "/home/reddy/soft/CODES/eDMFT-intel/bin/imp2lattc.py", line 69, in <listcomp>
    ndc=[ nd[a-1] for a in imp2latt[ic] ]
IndexError: list index out of range

Here is my .bashrc file, where the WIEN2k and compiled DMFT are added

# added by WIEN2k: BEGIN
# --------------------------------------------------------
alias lsi="ls -aslp *.in*"
alias lso="ls -aslp *.output*"
alias lsd="ls -aslp *.def"
alias lsc="ls -aslp *.clm*"
alias lss="ls -aslp *.scf* */*.scf"
alias lse="ls -aslp *.error"
alias LS="ls -aslp | grep /"
alias pslapw="ps -ef |grep "lapw""
alias cdw="cd /home/reddy/WIEN2k"
if [ "$OMP_NUM_THREADS" = "" ]; then export OMP_NUM_THREADS=4; fi
#export LD_LIBRARY_PATH=.....
export EDITOR="xterm -e vi"
export SCRATCH=./
if [ "$WIENROOT" = "" ]; then export WIENROOT=/home/reddy/soft/CODES/Wien2k/23.2; fi
export W2WEB_CASE_BASEDIR=/home/reddy/WIEN2k
export STRUCTEDIT_PATH=$WIENROOT/SRC_structeditor/bin
export PDFREADER=okular
export PATH=$WIENROOT:$STRUCTEDIT_PATH:$WIENROOT/SRC_IRelast/script-elastic:$PATH:.
export OCTAVE_EXEC_PATH=${PATH}::
export OCTAVE_PATH=${STRUCTEDIT_PATH}::

export PATH=$PATH:$WIENROOT:.
ulimit -s unlimited
alias octave="octave -p $OCTAVE_PATH"
# --------------------------------------------------------
# added by WIEN2k: END 

#export WIEN_DMFT_ROOT=/home/reddy/soft/CODES/eDMFT_new/eDMFT-master/bin
export WIEN_DMFT_ROOT=/home/reddy/soft/CODES/eDMFT-intel/bin
export PYTHONPATH=$PYTHONPATH:$WIEN_DMFT_ROOT
export WIENROOT=/home/reddy/soft/CODES/Wien2k/23.2
export SCRATCH="./"
export EDITOR="xterm -e vi"
export PATH=$WIENROOT:$WIEN_DMFT_ROOT:$PATH

Please let me know is there any other way to make the parallel job to work properly.

Thank you

with regards,
P. V. Sreenivasa Reddy,
NTU, Taiwan

**Comment by RayAmini** [comment]
It is difficult to say immediately what causes the problem. Are you sure you run the calculation from a designated dmft directory? Did you do dmft_copy.py? 

Could you run the parallell code for any of the tutorials, for which you know that the input files are correct?
